### PR TITLE
support pulling private images from docker pull

### PR DESCRIPTION
**Pull request by vieux**
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

### CODE REVIEW

**Comment by vieux** [comment]
ping @aluzzardi

File: api/handlers.go [hunk 1]
```diff
@@ -251,6 +252,11 @@ func postImagesCreate(c *context, w http.ResponseWriter, r *http.Request) {
 	wf := NewWriteFlusher(w)
 
 	if image := r.Form.Get("fromImage"); image != "" { //pull
+		authConfig := dockerclient.AuthConfig{}
+		buf, err := base64.URLEncoding.DecodeString(r.Header.Get("X-Registry-Auth"))
```
Comment (aluzzardi) [inline api/handlers.go]: This looks good, however I was wondering why passing the auth only when using `fromImage`?

If we receive a `X-Registry-Auth` from the client, shouldn't we always pass it?

Comment (vieux) [inline api/handlers.go]: when `fromImage` is not provided it means import, and we don't need `authConfig` for import

**Comment by aluzzardi** [comment]
LGTM

**Pull request merged**